ls1intum/ArTEMiS
# Feature/programming exercise instructor status

### Checklist

### Motivation and Context
It would be great if it was more obvious for the instructor if there is a problem with the solution/template repository.

### Description
Created a new component that shows a warning for solution/template repositories:
- template: One or more test cases are fulfilled
- solution: One or more test cases are NOT fulfilled

The component is displayed in the programming exercise table (course administration) and in the view page.
To show the component on the table I needed to add a join on the max result on the all programming-exercises for course method.

### Steps for Testing

1. Log in to ArTEMiS
2. Navigate to Course Administration
3. Create a programming exercise that has an issue with the repositories (see description above)
4. Check that the warnings appear in view/table

### Screenshots
View:
![image](https://user-images.githubusercontent.com/28230611/57371261-9924cb00-7192-11e9-983d-bda341e31e26.png)

Table:
![image](https://user-images.githubusercontent.com/28230611/57371863-30d6e900-7194-11e9-8b46-89b648e0fa79.png)
